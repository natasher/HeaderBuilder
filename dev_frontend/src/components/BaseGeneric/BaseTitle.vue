<template>
  <span class="title">{{ title }}</span>
</template>

<script>
export default {

  name: 'BaseTitle',

  props: {
    /**
     * Title of the elements to be shown.
     */
    title: {
      type    : String,
      required: true,
    },
  },

}
</script>
