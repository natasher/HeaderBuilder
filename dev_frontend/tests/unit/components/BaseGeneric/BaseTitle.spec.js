import { shallowMount } from '@vue/test-utils'
import BaseTitle from '../../../../src/components/BaseGeneric/BaseTitle.vue'

describe('BaseTitle.vue', () => {

  test('should render text passes in props', () => {
    const title   = 'My Awesome Title'
    const wrapper = shallowMount(BaseTitle, {
      propsData: {
        title,
      }
    })

    expect( wrapper.text() ).toBe( title )
  })

  test('should match snapshot', () => {
    const wrapper = shallowMount(BaseTitle, {
      propsData: {
        title: 'My Awesome Title'
      }
    })

    expect( wrapper.element ).toMatchSnapshot()
  })
})