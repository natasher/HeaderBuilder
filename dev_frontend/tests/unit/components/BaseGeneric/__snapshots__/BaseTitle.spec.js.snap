// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTitle.vue should match snapshot 1`] = `
<span
  class="title"
>
  My Awesome Title
</span>
`;
